lib/keychain/sec.rb

```diff
@@ -166,6 +162,40 @@ def load_attributes
       cf_dict = CF::Base.typecast(result.read_pointer).release_on_gc
       update_self_from_dictionary(cf_dict)
     end
+
+    def build_new_attributes
+      new_attributes = CF::Dictionary.mutable
+      @attributes.each do |key, value|
+        next unless self.class::ATTR_UPDATABLE.include?(key)
+        next if key == :klass && self.persisted?
+        key_cf = self.class::INVERSE_ATTR_MAP[key]
+        new_attributes[key_cf] = value.to_cf
+      end
+      new_attributes
+    end
+
+    def update
+      status = Sec.SecItemUpdate({Sec::Query::SEARCH_LIST => [self.keychain],
+                                  Sec::Query::ITEM_LIST => [self],
+                                  Sec::Query::CLASS => klass}.to_cf, build_new_attributes)
+      Sec.check_osstatus(status)
+
+      result = FFI::MemoryPointer.new :pointer
+      query = build_refresh_query
+      status = Sec.SecItemCopyMatching(query, result)
+      Sec.check_osstatus(status)
+      cf_dict = CF::Base.typecast(result.read_pointer)
```

Comment:
I think this is another missing release (although it looks like this was in the original code too)